### PR TITLE
Remove the deprecated `Plugin` class implementation, use the initalizer instead

### PR DESCRIPTION
**Pull request by res0nance**
Rewrite plugin to use modern constructs

### CODE REVIEW

File: src/main/java/hudson/plugins/extendedread/ExtendedReadPermissionEnabler.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package hudson.plugins.extendedread;
+
+import hudson.Extension;
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+import hudson.model.Item;
+
+@Extension
+public class ExtendedReadPermissionEnabler {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/extendedread/ExtendedReadPermissionEnabler.java]: Add `@Restricted(NoExternalUse.class)` just in case?

**Comment by res0nance** [comment]
@batmat it seems to be an infra failure with deleting windows directories. Seen it quite often recently

**Comment by oleg-nenashev** [comment]
It worked fine on my laptop last week, so I would suggest to go ahead and to release a new  major version. Will do it tomorrow if everybody is onboard

**Comment by batmat** [comment]
Agreed. Let's move on while infra issues are being worked on.

**Pull request merged**